src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -132,34 +146,57 @@
      * The client secret from the OpenShift server.
      */
     private final Secret clientSecret;
+    
+    /**
+     * This map contains defaults gathered for this objects fields when those fields were not set in the UI.
+     * We don't store our inferred defaults in those fields, because if we do, they will show up in the gui
+     * When the getters are called.
+     */
+    private final Map<String,String> overrides = new HashMap<String,String>();
+    
 
     @DataBoundConstructor
     public OpenShiftOAuth2SecurityRealm(String serviceAccountDirectory, String serviceAccountName, String serverPrefix, String clientId, String clientSecret) throws IOException, GeneralSecurityException {
         HttpTransport transport = HTTP_TRANSPORT;
+        
+        LOGGER.fine(String.format("ctor args sa dir %s sa name %s svr prefix %s client id %s client secret %s", serviceAccountDirectory, serviceAccountName, serverPrefix, clientId, clientSecret));
 
-        this.serviceAccountDirectory = Util.fixEmpty(serviceAccountDirectory);
-    	if (Util.fixEmpty(serviceAccountName) == null)
-    		this.serviceAccountName = "jenkins";
-    	else
-    		this.serviceAccountName = serviceAccountName;
-    	// TODO: should this be in a different method?  Will this show up in settings page?
-    	if (this.serviceAccountDirectory != null) {
-    		if (Util.fixEmpty(clientSecret) == null) {
-	            BufferedReader r = new BufferedReader(new FileReader(new File(this.serviceAccountDirectory, "token")));
-	            clientSecret = r.readLine();
+        String fixedServiceAccountDirectory = Util.fixEmpty(serviceAccountDirectory);
+        String fixedClientId = Util.fixEmpty(clientId);
+        String fixedClientSecret = Util.fixEmpty(clientSecret);
+        String fixedServerPrefix = Util.fixEmpty(serverPrefix);
+        
+        // decided not to default this since it is not even consistent within OpenShift;
+        // going through the example instantapp templates, the SA will be "default";
+        // going through the auto generated server, the SA will be "jenkins"
+        this.serviceAccountName = Util.fixEmpty(serviceAccountName);
```

Comment:
I've tried a few permuations:
- list users
- get user "~"
- get user "test" (where I logged in as test previously)

And I keep getting refused because the service account does not have permission to get users.

Typically, we grant the serviceaccount jenkins is running under the "edit" role.

I've even gone as far as adding the "sudoer" role to the serviceaccount, but something is still not right.

What aspect(s) do I need to change?